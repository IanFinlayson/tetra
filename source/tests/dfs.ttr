# print a maze to the screen
def printMaze(maze [[string]]):
    for line in maze:
        for char in line:
            print(char)
        print("\n")

# search the maze using recursive DFS
def search(maze [[string]], x int, y int) bool:
    maze[y][x] = "*"
    printMaze(maze)
    print("We are at (", x, ", ", y, ")\n")

    # check if we are done
    if x == 17 and y == 7:
        return true
    
    # mark this spot as seen
    maze[y][x] = "X"
    
    # if we haven't gone left, and not out of bounds, go left
    if ((x - 1) >= 0) and (maze[y][x - 1] == "-"):
        if search(maze, x - 1, y):
            return true

    # go bottom
    if ((y + 1) <= 7) and (maze[y + 1][x] == "-"):
        if search(maze, x, y + 1):
            return true

    # go top
    if ((y - 1) >= 0) and (maze[y - 1][x] == "-"):
        if search(maze, x, y - 1):
            return true

    # go right
    if ((x + 1) <= 17) and (maze[y][x + 1] == "-"):
        if search(maze, x + 1, y):
            return true

    # if we got here, there was no way on this path
    return false
  
 
def main():
    # this is extremely ugly
    maze = [["-", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#", "#"], \
            ["-", "-", "-", "-", "-", "-", "-", "-", "-", "-", "#", "#", "-", "-", "-", "-", "-", "#"], \
            ["-", "#", "#", "#", "#", "#", "#", "#", "#", "-", "#", "#", "-", "#", "#", "-", "#", "#"], \
            ["-", "#", "#", "#", "#", "#", "#", "#", "#", "-", "-", "-", "-", "#", "#", "-", "#", "#"], \
            ["-", "#", "#", "#", "#", "#", "#", "#", "#", "-", "#", "#", "#", "#", "#", "-", "#", "#"], \
            ["-", "-", "-", "-", "-", "#", "#", "#", "#", "-", "#", "#", "#", "#", "#", "-", "#", "#"], \
            ["#", "#", "#", "#", "#", "#", "#", "#", "#", "-", "#", "#", "#", "-", "-", "-", "#", "#"], \
            ["#", "#", "#", "#", "-", "-", "-", "-", "-", "-", "#", "#", "#", "#", "#", "-", "-", "-"]]

    if search(maze, 0, 0):
        print("Done!\n")
    else:
        print("No Path\n")

