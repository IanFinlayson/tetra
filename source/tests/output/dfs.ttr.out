*#################
----------##-----#
-########-##-##-##
-########----##-##
-########-#####-##
-----####-#####-##
#########-###---##
####------#####---
We are at (0, 0)
X#################
*---------##-----#
-########-##-##-##
-########----##-##
-########-#####-##
-----####-#####-##
#########-###---##
####------#####---
We are at (0, 1)
X#################
X---------##-----#
*########-##-##-##
-########----##-##
-########-#####-##
-----####-#####-##
#########-###---##
####------#####---
We are at (0, 2)
X#################
X---------##-----#
X########-##-##-##
*########----##-##
-########-#####-##
-----####-#####-##
#########-###---##
####------#####---
We are at (0, 3)
X#################
X---------##-----#
X########-##-##-##
X########----##-##
*########-#####-##
-----####-#####-##
#########-###---##
####------#####---
We are at (0, 4)
X#################
X---------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
*----####-#####-##
#########-###---##
####------#####---
We are at (0, 5)
X#################
X---------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
X*---####-#####-##
#########-###---##
####------#####---
We are at (1, 5)
X#################
X---------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XX*--####-#####-##
#########-###---##
####------#####---
We are at (2, 5)
X#################
X---------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXX*-####-#####-##
#########-###---##
####------#####---
We are at (3, 5)
X#################
X---------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXX*####-#####-##
#########-###---##
####------#####---
We are at (4, 5)
X#################
X*--------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (1, 1)
X#################
XX*-------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (2, 1)
X#################
XXX*------##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (3, 1)
X#################
XXXX*-----##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (4, 1)
X#################
XXXXX*----##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (5, 1)
X#################
XXXXXX*---##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (6, 1)
X#################
XXXXXXX*--##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (7, 1)
X#################
XXXXXXXX*-##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (8, 1)
X#################
XXXXXXXXX*##-----#
X########-##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (9, 1)
X#################
XXXXXXXXXX##-----#
X########*##-##-##
X########----##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (9, 2)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########*---##-##
X########-#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (9, 3)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########*#####-##
XXXXX####-#####-##
#########-###---##
####------#####---
We are at (9, 4)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####*#####-##
#########-###---##
####------#####---
We are at (9, 5)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########*###---##
####------#####---
We are at (9, 6)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####-----*#####---
We are at (9, 7)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####----*X#####---
We are at (8, 7)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####---*XX#####---
We are at (7, 7)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####--*XXX#####---
We are at (6, 7)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####-*XXXX#####---
We are at (5, 7)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X---##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####*XXXXX#####---
We are at (4, 7)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########X*--##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (10, 3)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########XX*-##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (11, 3)
X#################
XXXXXXXXXX##-----#
X########X##-##-##
X########XXX*##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (12, 3)
X#################
XXXXXXXXXX##-----#
X########X##*##-##
X########XXXX##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (12, 2)
X#################
XXXXXXXXXX##*----#
X########X##X##-##
X########XXXX##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (12, 1)
X#################
XXXXXXXXXX##X*---#
X########X##X##-##
X########XXXX##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (13, 1)
X#################
XXXXXXXXXX##XX*--#
X########X##X##-##
X########XXXX##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (14, 1)
X#################
XXXXXXXXXX##XXX*-#
X########X##X##-##
X########XXXX##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (15, 1)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##*##
X########XXXX##-##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (15, 2)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##*##
X########X#####-##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (15, 3)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####*##
XXXXX####X#####-##
#########X###---##
####XXXXXX#####---
We are at (15, 4)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####*##
#########X###---##
####XXXXXX#####---
We are at (15, 5)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####X##
#########X###--*##
####XXXXXX#####---
We are at (15, 6)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####X##
#########X###-*X##
####XXXXXX#####---
We are at (14, 6)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####X##
#########X###*XX##
####XXXXXX#####---
We are at (13, 6)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####X##
#########X###XXX##
####XXXXXX#####*--
We are at (15, 7)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####X##
#########X###XXX##
####XXXXXX#####X*-
We are at (16, 7)
X#################
XXXXXXXXXX##XXXX-#
X########X##X##X##
X########XXXX##X##
X########X#####X##
XXXXX####X#####X##
#########X###XXX##
####XXXXXX#####XX*
We are at (17, 7)
Done!

